[mipi_spi] Document new options (#5076)
OttoWinter/esphomedocs: components/display/mipi_spi.rst

- **use_axis_flips** (*Optional*): If true, the driver will use alternate bits in the MADCTL register to implement x and y mirroring. Defaults to false.
- **byte_order** (*Optional*): The byte order of the display buffer. Options are ``big_endian`` (default) and ``little_endian``. This affects the byte order for the buffer when
  using 16 bit color depth. The default is appropriate for the majority of displays.

**Note:** The maximum achievable data rate will depend on the chip type (e.g. ESP32 vs ESP32-S3) the pins used (on ESP32 using the default SPI pins allows higher rates) and the connection type (on-board connections will support higher rates than long cables or DuPont wires.) If in doubt, start with a low speed and test higher rates to find what works. A MISO pin should preferably not be specified, as this will limit the maximum rate in some circumstances, and is not required if the SPI bus is used only for the display.

Buffer Size
------------

The display driver writes data from a buffer to the display chip. When using :doc:`/components/lvgl/index` no buffer is required in the display driver itself, as LVGL will
allocate and use its own buffer.

> When not using LVGL, i.e. using a lambda function to update the display, a buffer is required to be allocated by the display driver.

Reviewer: Just avoiding these "i.e.", "e.g." and so on
```suggestion
When instead using a lambda function to update the display (and not LVGL), a buffer is required to be allocated by the display driver.
```

The size of this buffer is determined by the ``buffer_size`` option. The default is ``100%`` when PSRAM is configured, and otherwise will be calculated to
achieve a buffer size less than 20K bytes. For example a display of size 320x240 will have a buffer size of ``320 * 240 * 2`` bytes (for RGB565) = ``153600`` bytes.
If the buffer size is set to ``50%``, then the buffer would occupy ``76800`` bytes. If 8 bit color depth is used, then each pixel occupies only 1 byte.

Effect on Drawing Performance
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The buffer size is a trade-off between the size of the buffer and the performance of the display driver. A larger buffer size will provide better performance,
but on boards with limited memory, a smaller buffer size may be required to avoid running out of memory. When using a buffer less than 100%, the driver will

> call the drawing lambda multiple times to draw each chunk of the display, e.g. with 25% buffer size, the driver will call the drawing lambda 4 times to draw the display.

Reviewer: ```suggestion
call the drawing lambda multiple times to draw each chunk of the display. For example, with a 25% buffer size, the driver will call the drawing lambda four times to draw the display.
```
